frontend/src/pages/Match/index.js

```diff
@@ -33,29 +33,25 @@ function MatchPage() {
   const alertCtx = useContext(AlertContext);
   const userCtx = useContext(UserContext);
   // Difficulty
-  const [difficulty, setDifficulty] = useState('');
+  const difficulty = location.state?.difficulty;
   // Socket
-  const [socket, setSocket] = useState(null);
+  const socket = io(URL_MATCHING_SVC_CONNECT, {
+    path: '/api/matching-service/socket',
+  });
   const [room, setRoom] = useState('');
   const [isMatchFailed, setIsMatchFailed] = useState(false);
   // Timer
   const [counter, setCounter] = useState(DURATION);
   const [timerEnd, setTimerEnd] = useState(false);
 
-  // useEffect
   // Difficulty
   useEffect(() => {
-    if (location.state?.difficulty) {
-      setDifficulty(location.state.difficulty);
-    } else {
+    if (!difficulty) {
       navigate('/home');
       alertCtx.onShow('Please select a difficulty level!');
     }
-  }, [location.state, alertCtx, navigate]);
+  }, [difficulty, alertCtx, navigate]);
```

Comment:
have you check if these dependency array will cause infinite component rendering using `console.log`? If done, please ignore this.